### ISSUE TITLE
Visual glitches when scrolling while calling nvim_buf_set_lines on a floating window

### ISSUE THREAD

**Issue by lincheney** [neovim/neovim#20833]
### Neovim version (nvim -v)

NVIM v0.8.0 release

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Arch Linux 

### Terminal name/version

tested on xterm (374), xfce4-terminal 1.0.4, kitty (0.26.3), alacritty 0.10.1, 

### $TERM environment variable

xterm-256color

### Installation

system package manager

### How to reproduce the issue

vim config file:
```lua
local bufnr = vim.api.nvim_create_buf(true, false)
local winid = vim.api.nvim_open_win(bufnr, false, {relative = 'win', width = 0, height = 1, row = 10, col = 0})

vim.api.nvim_create_autocmd('WinScrolled', {callback=function()
    vim.api.nvim_buf_set_lines(bufnr, 0, 1, false, {'123'})
end})
```

Also create a text file to open by running: `seq 100 > file`.
I think this needs to be bigger than the terminal height.

Then open the file: `nvim --clean -u init.lua file`
Scroll down past the bottom using the mouse or by pressing `<c-e>`

### Expected behavior

The file displays normally, and the usual blank space is shown after the end of the file.
![Screenshot_2022-10-28_01-35-25](https://user-images.githubusercontent.com/1336117/198317645-d212591e-12dd-42c1-98bb-645ca185dbc8.png)

### Actual behavior

The last few lines of the file are displayed repeatedly
![Screenshot_2022-10-28_01-35-08](https://user-images.githubusercontent.com/1336117/198317447-33818260-c4e9-406a-af7c-82d46519c590.png)

**Comment by lewis6991** [comment]
I think this might have already been addressed via #20645 #20665 which will be available in v0.8.

**Comment by zeertzjq** [comment]
The steps lead to an error:
```
E5113: Error while calling lua chunk: /tmp/tmp.lua:2: 'width' key must be a positive Integer
```

**Comment by zeertzjq** [comment]
This look already fixed.

**Comment by lincheney** [comment]
> The steps lead to an error:
> 
> ```
> E5113: Error while calling lua chunk: /tmp/tmp.lua:2: 'width' key must be a positive Integer
> ```
> 
> Did you swap `width` and `row`?

Yeah woops sorry I did



> I think this might have already been addressed via #20645 #20665 which will be available in v0.8.

and yep you're right, I just cloned from master and built from source and it worked there.
